From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Wed, 14 Jul 2020 22:58 +0100

Jitterentropy library must be built with compiler optimisation disabled
as it *relies* on this. The below change to the Makefile ensures that abuild
general settings do not override the -O0 setting in the Makefile.

This is an Alpine-specific change and so not submitted upstream - it was
however discussed with upstream:
https://github.com/smuellerDD/jitterentropy-library/issues/55

---

diff -aur a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,8 @@
 CC ?= gcc
 #Hardening
 CFLAGS ?= -fwrapv --param ssp-buffer-size=4 -fvisibility=hidden -fPIE -Wcast-align -Wmissing-field-initializers -Wshadow -Wswitch-enum
-CFLAGS +=-Wextra -Wall -pedantic -fPIC -O0 -fwrapv -Wconversion
+CFLAGS :=-Wextra -Wall -pedantic -fPIC -O0 -fwrapv -Wconversion
+CPPFLAGS :=
 LDFLAGS +=-Wl,-z,relro,-z,now -lpthread
 
 GCCVERSIONFORMAT := $(shell echo `$(CC) -dumpversion | sed 's/\./\n/g' | wc -l`)
